tests/runtime/test_caikit_health_probe.py

```diff
@@ -93,52 +98,71 @@ def temp_probe_config(*args, **kwargs):
         (TlsMode.MTLS_COMMON_CLIENT_CA, False),
         (TlsMode.MTLS_SEPARATE_CLIENT_CA, True),
         (TlsMode.MTLS_SEPARATE_CLIENT_CA, False),
+        # Tests for no "localhost" in the SAN list
+        (TlsMode.TLS_NO_LOCALHOST, False),
+        (TlsMode.MTLS_NO_LOCALHOST, False),
     ],
 )
 @pytest.mark.parametrize("server_mode", ServerMode.__members__.values())
 def test_health_probe(test_config: Tuple[TlsMode, bool], server_mode: ServerMode):
     """Test all of the different ways that the servers could be running"""
     tls_mode, inline = test_config
-
-    # Get ports for both servers
-    http_port = tls_test_tools.open_port()
-    grpc_port = tls_test_tools.open_port()
-
-    # Set up tls values if needed
-    tls = tls_mode == TlsMode.TLS
-    mtls = tls_mode in [TlsMode.MTLS_COMMON_CLIENT_CA, TlsMode.MTLS_SEPARATE_CLIENT_CA]
-    with generate_tls_configs(
-        port=http_port,
-        tls=tls,
-        mtls=mtls,
-        inline=inline,
-        separate_client_ca=tls_mode == TlsMode.MTLS_SEPARATE_CLIENT_CA,
-    ) as config_overrides:
-        with temp_probe_config(
-            {
-                "runtime": {
-                    "grpc": {
-                        "port": grpc_port,
-                        "enabled": server_mode in [ServerMode.GRPC, ServerMode.BOTH],
-                    },
-                    "http": {
-                        "enabled": server_mode in [ServerMode.HTTP, ServerMode.BOTH],
-                    },
-                }
-            },
-            "merge",
-        ):
-            # Health probe fails with no servers booted
-            assert not caikit_health_probe.health_probe()
-            # If booting the gRPC server, do so
-            with maybe_runtime_grpc_test_server(grpc_port):
-                # If only running gRPC, health probe should pass
-                assert caikit_health_probe.health_probe() == (
-                    server_mode == ServerMode.GRPC
-                )
-                # If booting the HTTP server, do so
-                with maybe_runtime_http_test_server(
-                    http_port, tls_config_override=config_overrides
-                ):
-                    # Probe should always pass with both possible servers up
-                    assert caikit_health_probe.health_probe()
+    with alog.ContextLog(
+        log.info,
+        "---LOG CONFIG: tls: %s, inline: %s, server: %s---",
+        tls_mode.name,
+        inline,
+        server_mode.name,
+    ):
+        # Get ports for both servers
+        http_port = tls_test_tools.open_port()
+        grpc_port = tls_test_tools.open_port()
+
+        # Set up SAN lists if not putting "localhost" in
+        server_sans, client_sans = None, None
+        if "NO_LOCALHOST" in tls_mode.name:
+            server_sans = ["foo.bar"]
+            client_sans = ["baz.bat"]
+
+        # Set up tls values if needed
+        tls = tls_mode == TlsMode.TLS
```

Comment:
Since there's now a `TlsMode.TLS_NO_LOCALHOST` does this need to be
```
tls = tls_mode.name.startswith("TLS")
```

Reply:
Good find!